### ISSUE TITLE
[java] ExceptionAsFlowControl false positive when not catching the thrown exception

### ISSUE THREAD

**Issue by chrislake** [pmd/pmd#5489]
**Affects PMD Version:**
7.7.0

**Rule:**
[ExceptionAsFlowControl](https://pmd.github.io/pmd/pmd_rules_java_design.html#exceptionasflowcontrol)

**Description:**
A method which selectively throws exceptions and catches others appears to trigger the violation. Note that this only occurs when the exceptions are in their own class files. If everything is contained within the one class, the PMD warning is not shown.

**Code Sample demonstrating the issue:**

```java
CustomException.java
  public class CustomException extends Exception {
    // Usual exception handling stuff
  }

CustomApplicationException.java
  public class CustomApplicationException extends ApplicationException {
    // Usual exception handling stuff
  }

ApplicationException.java
  public class ApplicationException extends Exception {
    // Usual exception handling stuff
  }

MyObject.java
  private List getMyList() throws CustomException {
    try {
      List<String> theList = getList();
      if (theList.isEmpty()) {
        throw new CustomException("Unable to get list of stuff");
      }
      return theList;
    }
    catch (CustomApplicationException e) {  // ExceptionAsFlowControl: Exception thrown at line 5 is caught in this block.
      // Despite what PMD says, CustomException cannot be caught here
      throw new CustomException("A really bad exception occurred getting the list", e);
    }

  private List<String> getList() throws CustomApplicationException {
    // do real work to create the list which might cause this exception. This is really elsewhere in the codebase
    return new ArrayList<String>();
  }
}
```

**Expected outcome:**

PMD reports a violation at line 9, but that's wrong. That's a false positive.

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*
Maven and Eclipse.

**Comment by chrislake** [comment]
Closing this issue.

I was investigating another false positive and found the cause of it and this to be the fault in the maven configuration (`<typeResolution>false</typeResolution>`), which I think was left behind from when we used to do our compile and PMD all in the same mvn call. This seems to have matched Eclipse' warnings with it's configuration setting to match (`Enable using Java Project Build Path...` unchecked).

With these options correctly set, the PMD warning are no longer shown.
